picoquic/util.c

```diff
@@ -487,8 +490,15 @@ FILE * picoquic_file_open_ex(char const * file_name, char const * flags, int * l
 
     return F;
 }
-
 FILE* picoquic_file_open(char const* file_name, char const* flags)
 {
     return picoquic_file_open_ex(file_name, flags, NULL);
-}
\ No newline at end of file
+}
+
+/* Safely close files in a portable way */
+void picoquic_file_close(FILE * F)
+{
+    if (F != NULL) {
+        (void)fclose(F);
+    }
+}
```

Comment:
I wonder whether this should return `FILE * NULL`, so we can then write`F = picoquic_file_close(F)`, instead of
~~~
picoquic_file_close(F)
F = NULL
~~~
Or maybe use a macro:
~~~
#define PICOQUIC_FILE_CLOSE(F) {if(F != NULL){ fclose(F); F=NULL;}}
~~~